src/core/Akka/Pattern/BackoffSupervisor.cs

```diff
@@ -68,127 +90,155 @@ public sealed class StartChild : IDeadLetterSuppression
             private StartChild() { }
         }
 
-        /// <summary>
-        /// TBD
-        /// </summary>
         [Serializable]
         public sealed class ResetRestartCount : IDeadLetterSuppression
         {
-            /// <summary>
-            /// TBD
-            /// </summary>
-            public readonly int Current;
-
-            /// <summary>
-            /// TBD
-            /// </summary>
-            /// <param name="current">TBD</param>
-            /// <returns>TBD</returns>
             public ResetRestartCount(int current)
             {
                 Current = current;
             }
+
+            public int Current { get; }
         }
 
         #endregion
 
-        private readonly Props _childProps;
-        private readonly string _childName;
         private readonly TimeSpan _minBackoff;
         private readonly TimeSpan _maxBackoff;
         private readonly double _randomFactor;
+        private readonly SupervisorStrategy _strategy;
+
+        public BackoffSupervisor(
+            Props childProps,
+            string childName,
+            TimeSpan minBackoff,
+            TimeSpan maxBackoff,
+            IBackoffReset reset,
+            double randomFactor,
+            SupervisorStrategy strategy) : base(childProps, childName, reset)
+        {
+            _minBackoff = minBackoff;
+            _maxBackoff = maxBackoff;
+            _randomFactor = randomFactor;
+            _strategy = strategy;
+        }
 
-        private int _restartCount = 0;
-        private IActorRef _child = null;
+        protected override SupervisorStrategy SupervisorStrategy()
+        {
+            var oneForOne = _strategy as OneForOneStrategy;
```

Comment:
Scala introduced the code for OneForOneStrategy only for binary compatibility. If we don't need it, we could just return _strategy